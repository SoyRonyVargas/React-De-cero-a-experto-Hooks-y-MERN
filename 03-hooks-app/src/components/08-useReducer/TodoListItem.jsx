import React from 'react'

const TodoListItem = ({ id , index , desc , handleDelete }) => {
    return (
        <li key={id} className={"d-flex align-items-center mb-2"}>
            <p className='todo_paragraph mb-0 mx-3'>
                {index + 1} - {desc}
            </p>
            <div className='button-group btn-group align'>
                {/* <button className='btn btn-success'>
                                        <i className='fa fa-check'></i>
                                        Agregar
                                    </button> */}
                <button id='delete_todo' onClick={ () => handleDelete(id) } className='btn btn-danger btn-small btn-sm'>
                    Eliminar
                </button>
            </div>
        </li>
    )
}

export default TodoListItem