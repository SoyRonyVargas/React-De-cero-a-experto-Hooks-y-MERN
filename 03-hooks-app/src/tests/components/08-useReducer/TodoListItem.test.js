import TodoListItem from "../../../components/08-useReducer/TodoListItem"
import { demoTodos } from "../../fixtures"
import { shallow } from "enzyme"

describe('Probando el componente de <TodoListItem/>', () => { 

    const todo = demoTodos[0]
    
    const handleDelete = jest.fn()

    test('deberia mostrarse correctamente', () => { 

        const wrapper = shallow(
           <TodoListItem 
                handleDelete={handleDelete}
                index={0}
                {...todo}
           />
        )

        expect(wrapper).toMatchSnapshot()

    })

    test('debe de mostrar la funcion handleDelete', () => { 

        const wrapper = shallow(
            <TodoListItem 
                handleDelete={handleDelete}
                index={0}
                {...todo}
            />
        )

        wrapper.find("#delete_todo").simulate("click")

        expect(handleDelete).toHaveBeenCalled()
        
        expect(handleDelete).toHaveBeenCalledWith(todo.id)
        
        expect(handleDelete).toHaveBeenCalledTimes(1)

    })
    
    test('debe de mostrar el texto correctamente del componente', () => { 

        const wrapper = shallow(
            <TodoListItem 
                handleDelete={handleDelete}
                index={0}
                {...todo}
            />
        )

        const paragraph = wrapper.find(".todo_paragraph").first()

        expect(paragraph.text()).toBe(`1 - ${ todo.desc}`)

    })

})