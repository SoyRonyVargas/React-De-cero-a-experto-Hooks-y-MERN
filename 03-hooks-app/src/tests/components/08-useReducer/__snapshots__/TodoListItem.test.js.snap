// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando el componente de <TodoListItem/> deberia mostrarse correctamente 1`] = `
<li
  className="d-flex align-items-center mb-2"
  key="0"
>
  <p
    className="todo_paragraph mb-0 mx-3"
  >
    1
     - 
    Aprender React âš›
  </p>
  <div
    className="button-group btn-group align"
  >
    <button
      className="btn btn-danger btn-small btn-sm"
      id="delete_todo"
      onClick={[Function]}
    >
      Eliminar
    </button>
  </div>
</li>
`;
